Core/src/main/java/org/gusdb/fgputil/DualBufferBinaryRecordReader.java

```diff
@@ -47,23 +50,33 @@ public class DualBufferBinaryRecordReader implements OptionStream<byte[]>, AutoC
    * @param recordLength size of binary records in bytes
    * @param recordsPerBuffer number of records to store in a buffer at one time.  Note: memory
    * footprint will be approximately recordLength * recordsPerBuffer * 2.
+   * @param recordBufferSupplier lazily-provided buffer for storing contents of a record. This enables client to determine
+   * whether a single shared buffer is used, or a new buffer is instantiated per record.
    * @throws IOException if unable to read file
    */
-  public DualBufferBinaryRecordReader(Path file, int recordLength, int recordsPerBuffer) throws IOException {
+  public DualBufferBinaryRecordReader(Path file, int recordLength, int recordsPerBuffer,
+                                      Supplier<byte[]> recordBufferSupplier) throws IOException {
     _file = file;
     _exec = Executors.newSingleThreadExecutor();
     _channel = AsynchronousFileChannel.open(file, Set.of(StandardOpenOption.READ), _exec);
     _recordLength = recordLength;
     _bufferSize = recordLength * recordsPerBuffer;
-    _current = ByteBuffer.allocate(_bufferSize);
-    _next = ByteBuffer.allocate(_bufferSize);
+    _current = ByteBuffer.allocateDirect(_bufferSize);
```

Comment:
Cool.  Had to read up on direct buffers... :)

Reply:
Honestly, it doesn't make a huge difference. Especially when compared to the amount of time spent copying the bytes into the buffer.